vkubelet/apiserver.go

```diff
@@ -34,11 +37,58 @@ func ApiserverStart(provider Provider) {
 	r.HandleFunc("/exec/{namespace}/{pod}/{container}", ApiServerHandlerExec).Methods("POST")
 	r.NotFoundHandler = http.HandlerFunc(NotFound)
 
+	if metricsAddr != "" {
+		go MetricsServerStart(metricsAddr)
```

Comment:
can we move the Metrics Server into a new file, and start it from the vkubelet.go?

Reply:
The reason I didn't do this for now is right now `ApiServerStart` is setting a global (for provider) and these will race with each other.
I'd rather refactor this in a separate PR.